src/QueryToMany.php

```diff
@@ -170,20 +170,12 @@ private function getRelation(): EloquentRelation
         if ($relation instanceof EloquentHasMany ||
             $relation instanceof EloquentBelongsToMany ||
             $relation instanceof EloquentHasManyThrough ||
-            $relation instanceof EloquentMorphMany
+            $relation instanceof EloquentMorphMany ||
+            $relation instanceof EloquentRelation
```

Comment:
So I think adding this as the final check is a bit pointless - presumably all of the previous classes it has checked in the `if` are instances of `EloquentRelation`.

I think we should _flip_ the logic.

If the relation is an instance of any of the _known_ to-one relations, it should throw saying the relation is not a to-many relation. Otherwise it can just accept the relation and assume it's to-many.

I believe the `QueryToOne` class will contain the list of known to-one relations, as it probably has a method that is the equivalent of this one.

If we're changing the logic in `QueryToMany`, we should also change the logic in `QueryToOne`... i.e. query-to-one should throw if the relation is an instance of any _known_ to many relations.